container_linux.go

```diff
@@ -777,3 +815,37 @@ func (c *linuxContainer) currentState() (*State, error) {
 	}
 	return state, nil
 }
+
+// orderNamespacePaths sorts that namespace paths into a list of paths that we
+// can safely setns to.
+func orderNamespacePaths(namespaces map[configs.NamespaceType]string) ([]string, error) {
+	paths := []string{}
+	for _, nsType := range []configs.NamespaceType{
+		// TODO: enable configs.NEWUSER,
+		configs.NEWIPC,
+		configs.NEWUTS,
+		configs.NEWNET,
+		configs.NEWPID,
+		// mnt namespace must be the last one. After switching to mnt
+		// namespace, the old /proc will be inaccessible.
```

Comment:
I think it's ok. But userns must be the last one, otherwise you will not have permissions to switch other namespaces.
https://github.com/torvalds/linux/blob/master/fs/namespace.c#L3235